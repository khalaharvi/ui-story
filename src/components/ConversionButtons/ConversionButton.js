import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { cx } from './ConversionButtons';

const ConversionButton = ({
  icon,
  type,
  onClick,
  label,
  url,
  attributes
}) => (
  <Fragment>
    {
      (url) ?
        <a
          href={url}
          className={classnames(cx('btn'), cx(`btn-${type}`))}
          onClick={onClick}
          {...attributes}
        >
          {icon}
        </a> :
        <button
          className={classnames(cx('btn'), cx(`btn-${type}`))}
          type="button"
          onClick={onClick}
          aria-label={label}
          {...attributes}
        >
          {icon}
        </button>
    }
  </Fragment>
);

ConversionButton.propTypes = {
  icon: PropTypes.element.isRequired,
  type: PropTypes.oneOf(['facebook', 'twitter', 'pinterest', 'email', 'print', 'download']).isRequired,
  onClick: PropTypes.func,
  label: PropTypes.string,
  url: PropTypes.string,
  attributes: PropTypes.object // eslint-disable-line react/forbid-prop-types
};

export default ConversionButton;
