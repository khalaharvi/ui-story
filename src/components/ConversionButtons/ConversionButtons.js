import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import {
  Facebook as FacebookIcon,
  Pinterest as PinterestIcon,
  Twitter as TwitterIcon,
  Email as EmailIcon,
  Print as PrintIcon,
  Download as DownloadIcon
} from 'icons';
import { createClassNames } from 'utils';
import ConversionButton from './ConversionButton';

export const cx = createClassNames('conversion-buttons');

const icons = {
  facebook: <FacebookIcon />,
  twitter: <TwitterIcon />,
  pinterest: <PinterestIcon />,
  email: <EmailIcon />,
  print: <PrintIcon />,
  download: <DownloadIcon />
};

const ConversionButtons = ({ buttons, className, share }) => (
  <div className={classnames(cx(''), className)}>
    {
      buttons.map(button => (
        <ConversionButton
          key={`dh-conversion-button-${button.type}`}
          icon={icons[button.type]}
          type={button.type}
          onClick={button.onClick ? e => button.onClick(e, share) : undefined}
          url={button.url}
          attributes={button.attributes}
        />
      ))
    }
  </div>
);

ConversionButtons.propTypes = {
  buttons: PropTypes.arrayOf(PropTypes.shape({
    type: PropTypes.oneOf(['facebook', 'twitter', 'pinterest', 'email', 'print', 'download']).isRequired,
    onClick: PropTypes.func,
    label: PropTypes.string,
    url: PropTypes.string,
    attributes: PropTypes.object // eslint-disable-line react/forbid-prop-types
  })),
  className: PropTypes.string,
  share: PropTypes.shape({
    contentId: PropTypes.number.isRequired,
    customerId: PropTypes.number.isRequired,
    content: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })
};

export default ConversionButtons;
