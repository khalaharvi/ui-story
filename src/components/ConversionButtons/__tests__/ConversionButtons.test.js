import React from 'react';
import { mount, shallow } from 'enzyme';

import ConversionButtons from '../ConversionButtons';

describe('<ConversionButtons />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ConversionButtons
        buttons={[
          {
            type: 'facebook'
          },
          {
            type: 'pinterest'
          },
          {
            type: 'twitter'
          },
          {
            type: 'print'
          },
          {
            type: 'email'
          },
          {
            type: 'download'
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with custom className', () => {
    const wrapper = shallow(
      <ConversionButtons
        className="my-conversion-buttons"
        buttons={[
          {
            type: 'facebook'
          },
          {
            type: 'pinterest'
          },
          {
            type: 'twitter'
          },
          {
            type: 'print'
          },
          {
            type: 'email'
          },
          {
            type: 'download'
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly with custom button's attributes", () => {
    const wrapper = shallow(
      <ConversionButtons
        buttons={[
          {
            type: 'facebook',
            attributes: {
              'data-attribute': 'attribute'
            }
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with share', () => {
    const wrapper = shallow(
      <ConversionButtons
        share={{
          contentId: 1,
          customerId: 2,
          content: 'content',
          description: 'description',
          image: 'image'
        }}
        buttons={[
          {
            type: 'facebook'
          },
          {
            type: 'pinterest'
          },
          {
            type: 'twitter'
          },
          {
            type: 'print'
          },
          {
            type: 'email'
          },
          {
            type: 'download'
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with custom onClick handler', () => {
    const onClick = jest.fn();
    const wrapper = mount(
      <ConversionButtons
        buttons={[
          {
            type: 'facebook',
            onClick
          },
          {
            type: 'pinterest'
          }
        ]}
      />
    );

    const buttons = wrapper.find('button');
    expect(buttons).toHaveLength(2);

    buttons.first().simulate('click');
    expect(onClick.mock.calls).toHaveLength(1);

    wrapper.unmount();
  });

  it("should pass 'share' prop to custom click handler as 2nd argument", () => {
    const onClick = jest.fn();
    const wrapper = mount(
      <ConversionButtons
        share={{
          contentId: 1,
          customerId: 2,
          content: 'content',
          description: 'description',
          image: 'image'
        }}
        buttons={[
          {
            type: 'facebook',
            onClick
          }
        ]}
      />
    );

    wrapper.find('button').simulate('click');

    expect(onClick.mock.calls).toHaveLength(1);
    expect(onClick.mock.calls[0][1]).toEqual({
      contentId: 1,
      customerId: 2,
      content: 'content',
      description: 'description',
      image: 'image'
    });

    wrapper.unmount();
  });
});
