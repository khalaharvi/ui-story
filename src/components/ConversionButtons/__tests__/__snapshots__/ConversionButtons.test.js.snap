// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConversionButtons /> renders correctly 1`] = `
<div
  className="dh-conversion-buttons"
>
  <ConversionButton
    icon={<Facebook />}
    key="dh-conversion-button-facebook"
    type="facebook"
  />
  <ConversionButton
    icon={<Pinterest />}
    key="dh-conversion-button-pinterest"
    type="pinterest"
  />
  <ConversionButton
    icon={<Twitter />}
    key="dh-conversion-button-twitter"
    type="twitter"
  />
  <ConversionButton
    icon={<Print />}
    key="dh-conversion-button-print"
    type="print"
  />
  <ConversionButton
    icon={<Email />}
    key="dh-conversion-button-email"
    type="email"
  />
  <ConversionButton
    icon={<Download />}
    key="dh-conversion-button-download"
    type="download"
  />
</div>
`;

exports[`<ConversionButtons /> renders correctly with custom button's attributes 1`] = `
<div
  className="dh-conversion-buttons"
>
  <ConversionButton
    attributes={
      Object {
        "data-attribute": "attribute",
      }
    }
    icon={<Facebook />}
    key="dh-conversion-button-facebook"
    type="facebook"
  />
</div>
`;

exports[`<ConversionButtons /> renders correctly with custom className 1`] = `
<div
  className="dh-conversion-buttons my-conversion-buttons"
>
  <ConversionButton
    icon={<Facebook />}
    key="dh-conversion-button-facebook"
    type="facebook"
  />
  <ConversionButton
    icon={<Pinterest />}
    key="dh-conversion-button-pinterest"
    type="pinterest"
  />
  <ConversionButton
    icon={<Twitter />}
    key="dh-conversion-button-twitter"
    type="twitter"
  />
  <ConversionButton
    icon={<Print />}
    key="dh-conversion-button-print"
    type="print"
  />
  <ConversionButton
    icon={<Email />}
    key="dh-conversion-button-email"
    type="email"
  />
  <ConversionButton
    icon={<Download />}
    key="dh-conversion-button-download"
    type="download"
  />
</div>
`;

exports[`<ConversionButtons /> renders correctly with share 1`] = `
<div
  className="dh-conversion-buttons"
>
  <ConversionButton
    icon={<Facebook />}
    key="dh-conversion-button-facebook"
    type="facebook"
  />
  <ConversionButton
    icon={<Pinterest />}
    key="dh-conversion-button-pinterest"
    type="pinterest"
  />
  <ConversionButton
    icon={<Twitter />}
    key="dh-conversion-button-twitter"
    type="twitter"
  />
  <ConversionButton
    icon={<Print />}
    key="dh-conversion-button-print"
    type="print"
  />
  <ConversionButton
    icon={<Email />}
    key="dh-conversion-button-email"
    type="email"
  />
  <ConversionButton
    icon={<Download />}
    key="dh-conversion-button-download"
    type="download"
  />
</div>
`;
